cpp/src/gandiva/jni/jni_common.cc

```diff
@@ -663,7 +665,17 @@ std::shared_ptr<arrow::Buffer> JavaSecondaryCache::Get(
   auto data = std::shared_ptr<arrow::Buffer>(
       new arrow::Buffer(reinterpret_cast<uint8_t*>(ret_address), ret_size));
 
-  return data;
+  // copy the buffer and release the original memory
+  auto result =
+      arrow::Buffer::CopyNonOwned(*data.get(), arrow::default_cpu_memory_manager());
```

Comment:
how is this memory released back?

Reply:
Since it is allocated in gandiva, shouldn't the memory be released on object destruction?